pkgs/_analyzer_cfe_macros/lib/metadata_converter.dart

```diff
@@ -275,24 +281,27 @@ T? convert<T>(Object? object) => switch (object) {
         as T,
   front_end.ConstructorInvocation o =>
     dart_model.ConstructorInvocation(
-          type: convertToTypeAnnotation(o.type),
+          type: convert(o.type),
```

Comment:
was this added manually?

Reply:
Yes, it's needed whenever the type is a union type, the script doesn't get this one right.

The script was not really intended for long term used, we need to build a maintainable solution :)